src/main/java/org/truffleruby/core/array/DelegatedArrayMirror.java

```diff
@@ -36,8 +36,17 @@ public void set(int index, Object value) {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public ArrayMirror newMirror(int newLength) {
+        ArrayMirror newMirror = storageMirror.newMirror(newLength);
+        DelegatedArrayStorage newStorage = new DelegatedArrayStorage(newMirror.getArray(), 0, newLength);
+        return new DelegatedArrayMirror(newStorage, newMirror);
+    }
```

Comment:
Since this makes a new storage and mirror, is there a reason to use a DelegatedArrayStorage over just returning `newMirror`?
The other implementations of newMirror don't seem to copy the original storage and so it seems implicitly separate since it's a new allocation.